server.js

```diff
@@ -1,34 +1,100 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+import express, { request } from "express";
+import ListEndpoints from "express-list-endpoints";
+import cors from "cors";
+
+import booksData from "./data/books.json";
+
+// Defines the port the app will run on. Defaults to 8080
+// PORT=9000 npm start
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(express.json());
+
+// ROUTES
+
+// Main root is '/'
+// ListEndPoints sending list of end points
+app.get("/", (req, res) => {
+  res.send(ListEndpoints(app));
+});
+
+// http://localhost:8080/books
+// http://localhost:8080/books?author=J.K. Rowling-Mary GrandPré
+// http://localhost:8080/books?title=Harry Potter and the Order of the Phoenix
+// .lower case will enable searches like:
+// http://localhost:8080/books?author=j.k.%20row (%20) is space in URL
+// query param, if if searching for specific auther or title
+// '/books' end point to get all books
+// res.json the data always convert data to string
+// with query params returning an array of objects
+
+app.get("/books", (req, res) => {
+  const { author, bookTitle } = req.query;
+
+  if (bookTitle) {
+    const booksTitle = booksData.filter((book) => {
+      return book.title.toLowerCase().includes(bookTitle.toLowerCase());
+    });
+    res.json(booksTitle);
+  } else if (author) {
+    const booksList = booksData.filter((book) => {
+      return book.authors.toLowerCase().includes(author.toLowerCase());
+    });
+    res.json(booksList);
+  }
+  res.send(booksData);
+});
+
+// http://localhost:8080/books/book/1
+// :id = param to trigger a param example console.log(req params)
+// it will look for the specific param, in this case a specific id from booksData
+// returns one specific object
+app.get("/books/book/:bookID", (req, res) => {
+  const { bookID } = req.params;
+  const book = booksData.find((b) => b.bookID === +bookID);
+  if (!book) {
+    res.status(404).send(`error, there  is no book with this ${bookID}`);
+  }
+  res.json(book);
+});
```

Comment:
When you filtering your data by its primal key you do not specify it in the endpoint address: `/books/:id` would be a proper address here.